### ISSUE TITLE
[SubPlat][MDN] Update MDN Plus Cancellation Exit Survey

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#12056]
Problem: Currently, all cancellation email Exit Surveys are specific to the Privacy and Security products, which does not serve all of our future products.

Request: In preparation for the MDN Plus release, please update Cancellation Emails for the MDN Plus product to the following URL: [https://survey.alchemer.com/s3/6760254/MDNPlusCancellation](https://survey.alchemer.com/s3/6760254/MDNPlusCancellation|smart-link) 

**Solution** add metadata in Stripe to url at the product level or default to the original.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4664)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/21202">Screen Shot 2022-03-03 at 10.46.14 AM.png</a>

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Reino Muhl : This ticket is similar to https://mozilla-hub.atlassian.net/browse/FXA-3589 ( https://mozilla-hub.atlassian.net/browse/FXA-3589|smart-link ) where a new button label was added for the button in the PaymentConfirmation component during checkout.

This case will be a little different, since it’s in an email on the auth server. In a way it may be a little easier, since we don’t have to worry about localizing the value.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

I added the qa+ label to this ticket. Please add STR once this closes for QA to verify in stage.

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Hello team,

Could youplease help us with the STR for ths issue, so that we can validate it against the Staging environment?

Thank you.

Reino Muhl Bianca Danforth 

CC: Maria Berlinger Madalin Cotetiu

**Comment by data-sync-user** [comment]
➤ Reino Muhl commented:

Hi Luminita Buruian 

Please see below. If you need more information, please let me know.

  1. Create an FxA Account and subscribe to the following 2 products via the web
    1. MDN Plus ( https://payments-stage.fxa.nonprod.cloudops.mozgcp.net/products/prod_Jtbg9tyGyLRuB0?device_id=f3acc4b843924f5ba9e0bf077bd49e9c&flow_begin_time=1647882443133&flow_id=ab879f6e9e6b4ad5d32e0b1e9839e4e730122762d5dca5ca64d1374f8e963fe3&plan=price_1K6X7gKb9q6OnNsLi44HdLcC&entrypoint=www.mozilla.org-vpn-product-page&form_type=button&service=e6eb0d1e856335fc&utm_source=www.mozilla.org-vpn-product-page&utm_medium=referral&utm_campaign=vpn-product-page&data_cta_position=pricing )
      1. Product: prod_Jtbg9tyGyLRuB0
      2. Plan: price_1K6X7gKb9q6OnNsLi44HdLcC

    2. Mozilla VPN ( https://payments-stage.fxa.nonprod.cloudops.mozgcp.net/products/prod_FiJ42WCzZNRSbS?device_id=763eff7537494fdf8e1538c0418695ea&flow_begin_time=1647882738322&flow_id=3a852f2191447b5296969ae3eda68f3ffbcda7c5b3046ccd5b5b9cadd13aa382&plan=price_1J0Y12Kb9q6OnNsL4SB2hhmp&entrypoint=www.mozilla.org-vpn-product-page&form_type=button&service=e6eb0d1e856335fc&utm_source=www.mozilla.org-vpn-product-page&utm_medium=referral&utm_campaign=vpn-product-page&data_cta_position=pricing )
      1. Product:  prod_FiJ42WCzZNRSbS
      2. plan: price_1J0Y12Kb9q6OnNsL4SB2hhmp

  2. Cancel the Subscription in each of the 3 ways
    1. In the Subscription Management page cancel the subscription
    2. Delete the FxA Account - which will cancel the subscription
    3. Failed payment - Which results in subscription cancellation (Not sure how to test this. Could discuss more if necessary)

  3. Review the email received when the subscription is cancelled.

Expected

Depending on the product, the cancellation survey mentioned in the email should be different.

  * MDN Plus
    * https://survey.alchemer.com/s3/6760254/MDNPlusCancellation ( https://survey.alchemer.com/s3/6760254/MDNPlusCancellation|smart-link ) 

  * Mozilla VPN
    * https://survey.alchemer.com/s3/6534408/Privacy-Security-Product-Cancellation-of-Service-Q4-21 ( https://survey.alchemer.com/s3/6534408/Privacy-Security-Product-Cancellation-of-Service-Q4-21|smart-link )

**Comment by data-sync-user** [comment]
➤ Maria Berlinger commented:

Hello,

We have tried the following scenarios:

1.We canceled the subscriptions from “Subscription Management”

  * The cancellation surveys are received correctly for both products;

2. We deleted the FxA account

  * The cancellation surveys are not received and in the Stripe dashboard the subscription is seen as active;
  * Even if we have deleted the account from Stripe dashboard the subscription remains active and the emails are not received;

3. We tried to trigger the failed payment scenario

  * Using credit cards from +here+ ( https://stripe.com/docs/testing#declined-payments ) the payment is not finalized, so we can’t check this scenario;
  * We also have tried to change the card number with one from +here+ ( https://stripe.com/docs/testing#declined-payments ) for Mozilla VPN Daily subscription in order to fail the next payment but the card can’t be changed to any special card used for testing failed payments;

If there are any other workarounds please let us know.

Thank you!

**Comment by data-sync-user** [comment]
➤ Reino Muhl commented:

HiMaria Berlinger 

My apologies for not getting back to you sooner.

For point number 2, this is actually due to an unrelated bug, which will be fixed as part of this ticket ( https://mozilla-hub.atlassian.net/browse/FXA-4764 ). Could we test this use case as part of that bug?

For point number 3, could you try either of these 2 options?

  1. Navigate to the FxA - Subscription Management page, and change the credit card to the following. (Note, this seems to be a new declined payments card that Stripe added within the last 2-3 weeks)
    1. 4000000000000341

  2. Would it be possible for the QA team to sign into Stripe Stage and delete the card from the customer?

**Comment by data-sync-user** [comment]
➤ Maria Berlinger commented:

Hello Reino Muhl ,

Sorry for the late response. 

We have tried both options for point number 3 and the following output has come:

  1. We have made a daily VPN subscription and changed the CC number to the one provided in your last comment. 

  * On the first day after the card was changed the “Payment failed” mail was received
  * On the second day, the “Payment failed” and the “Your Mozilla VPN subscription has been canceled” emails have been received, and the cancellation mail has the right survey link for Mozilla VPN. For the MDN product, we can not verify this since we do not have a daily subscription.

2. For another account where we have had a Mozilla VPN daily subscription and an MDN Plus monthly subscription and we deleted the CC from the stripe dashboard.

  * For the Mozilla VPN daily subscription on the first day after the card was deleted the “Payment failed” mail was received.
  * For the Mozilla VPN daily subscription on the second day after the card was deleted the “Payment failed” and the “Your Mozilla VPN subscription has been canceled” emails have been received, and the cancellation mail has the right survey link for Mozilla VPN. 
For the MDN product, we will verify if the cancellation email is received  for this scenario after the 16th of April.

**Comment by data-sync-user** [comment]
➤ Maria Berlinger commented:

Hello,

For this issue the following scenarios were verified:

>1.We canceled the subscriptions from “Subscription Management”

  * >The cancellation surveys are received correctly for both products;

This scenario was previously checked;

>2. We deleted the FxA account

  * >The cancellation surveys are not received and in the Stripe dashboard the subscription is seen as active;
  * >Even if we have deleted the account from Stripe dashboard the subscription remains active and the emails are not received;

This scenario was verified as part of +this ticket+ ( https://mozilla-hub.atlassian.net/browse/FXA-4764 ).

>We have tried both options for point number 3 and the following output has come:

  1. >We have made a daily VPN subscription and changed the CC number to the one provided in your last comment. 

  * >On the first day after the card was changed the “Payment failed” mail was received
  * >On the second day, the “Payment failed” and the “Your Mozilla VPN subscription has been >>canceled” emails have been received, and the cancellation mail has the right survey link for Mozilla VPN. For the MDN product, we can not verify this since we do not have a daily subscription.

2.> For another account where we have had a Mozilla VPN daily subscription and an MDN Plus monthly subscription and we deleted the CC from the stripe dashboard.

  * >For the Mozilla VPN daily subscription on the first day after the card was deleted the “Payment failed” mail was received.
  * >For the Mozilla VPN daily subscription on the second day after the card was deleted the “Payment failed” and the “Your Mozilla VPN subscription has been canceled” emails have been received, and the cancellation mail has the right survey link for Mozilla VPN. For the MDN product, we will verify if the cancellation email is received  for this scenario after the 16th of April.

For this scenario the “Your Mozilla VPN subscription has been cancelled“ email was received on 30th March when the cc was deleted.

For the MDN Plus subscription the “Payment failed” emails were received on the 16th and 17th of April and the “Your MDN Plus subscription has been cancelled“ mail was also received in the 17th of April

Based on the above, we are marking this as qa-verified.
